karate-tests/src/test/java/mocks/sandbox/supporting-functions.js

```diff
@@ -86,30 +85,74 @@ function isValidUUID (uuid) {
   return regex.test(uuid)
 }
 
+/*
+   * validate the oauth2 bearer token
+    */
+function isValidBearerToken (token) {
+  const tokenParts = token.split(' ')
+  if (tokenParts.length !== 2) {
+    return false
+  } else if (tokenParts[0] !== 'Bearer') {
+    return false
+  } else if (!isValidUUID(tokenParts[1])) {
+    return false
+  }
+  return true
+}
+
 /**
  * Validates the headers of a request.
  *
  * @param {Request} request - The request object.
  * @returns {boolean} - Returns true if the headers are valid, false otherwise.
  */
-// eslint-disable-next-line no-unused-vars
-function validateHeaders (request) {
-  const X_REQUEST_ID = 'X-Request-ID'
-
+function validateAuthHeader (request) {
   let valid = true
+  let diagnostics = ''
+  // Check if the Authorization header is present and correct
+  const authorization = request.header('Authorization')
+  if (!('authorization' in request.headers)) {
```

Comment:
Can we not check if the authorization const declared above is not null instead?

Reply:
Okay I've improved line 114 with the simpler:

`if (authorization === null) {`